models/migrationscripts/updateSchemas20220513.go

```diff
@@ -0,0 +1,72 @@
+package migrationscripts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/merico-dev/lake/models/migrationscripts/archived"
+	"gorm.io/gorm"
+)
+
+type RefsIssuesDiffs20220513 struct {
+	NewRefId        string `gorm:"primaryKey;type:varchar(255)"`
+	OldRefId        string `gorm:"primaryKey;type:varchar(255)"`
+	NewRefCommitSha string `gorm:"type:varchar(40)"`
+	OldRefCommitSha string `gorm:"type:varchar(40)"`
+	IssueNumber     string `gorm:"type:varchar(255)"`
+	IssueId         string `gorm:"primaryKey;type:varchar(255)"`
+	archived.NoPKModel
+}
+
+func (RefsIssuesDiffs20220513) TableName() string {
+	return "refs_issues_diffs"
+}
+
+type updateSchemas20220513 struct{}
+
+func (*updateSchemas20220513) Up(ctx context.Context, db *gorm.DB) error {
+	cursor, err := db.Model(archived.RefsIssuesDiffs{}).Rows()
+	if err != nil {
+		return err
+	}
+	defer cursor.Close()
+
+	duplicateData := make(map[string]interface{})
```

Comment:
the naming is confusing, this looks like `uniqueRows`  to me.

Reply:
how ever, the solution here is heavy on memory.
I think would be better to create a new table first, and try to insert all data into that table by ignoring duplicate record.
and then drop the old table, rename the new table .
with this approach, we dont need to keep track of anything in memory at all.